pkg/multiclusterservice/controller.go

```diff
@@ -105,7 +105,7 @@ func (c *Controller) runWorker() {
 			}
 
 			if !exists {
-				c.multiClusterServiceDeleted(key)
+				c.multiClusterServiceDeleted(obj, key)
```

Comment:
This won't work as `obj` will be nil in this case. We'll need to handle delete directly now instead of queueing.

Reply:
Done